lib/src/streams/sha384_web.dart

```diff
@@ -0,0 +1,67 @@
+@JS('SHA384')
+library SHA384;
+
+import 'dart:html';
+import 'dart:js_util';
+import 'dart:typed_data';
+import 'dart:async';
+
+import 'package:js/js.dart';
+
+import 'data_models.dart';
+
+@JS()
+@anonymous
+class IHasher {
+  external IHasher init();
+  external IHasher update(Uint8List data);
+  external dynamic digest(String outputType);
+  external dynamic save();
+  external IHasher load(dynamic data);
+  external int get blockSize;
+  external int get digestSize;
+}
+
+@JS('createSHA384')
+external Future<IHasher> _createSHA384();
+
+Future<IHasher> createSHA384() async {
+  return await promiseToFuture(_createSHA384());
+}
+
+@JS('sha384Hash')
+external Future<String> _sha384Hash(Uint8List data);
+
+class SHA384Hash implements SHA384Hasher {
+  late IHasher _hasher;
+
+  @override
+  init() async {
+    _hasher = await promiseToFuture(_createSHA384());
+  }
+
+  @override
+  update(data) {
+    _hasher.update(data);
+  }
+
+  @override
+  hash() async {
+    return Uint8List.fromList(_hasher.digest('binary'));
+  }
+}
+
+Future<Uint8List> sha384(Uint8List data) async {
+  return await promiseToFuture(_sha384Hash(data));
+}
+
+Uint8List _hexStringToUint8List(String hex) {
+  var length = hex.length;
+  var bytes = Uint8List(length ~/ 2);
```

Comment:
Will the input of this method take a hex with no even length?

Reply:
the string in this case (sha384 hash) will never be even, but we don't use this function anymore so I removed it
[`cb9fe81` (#52)](https://github.com/ardriveapp/arweave-dart/pull/52/commits/cb9fe812d8313e9eedb2285b4dea50e27f702200)